### ISSUE TITLE
[macOS] Building Development branch fails on MacBook 2020 machines due to failed CompileC LogFile_mac command

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#8132]
**Describe the bug**
CMake fails to build the latest Development branch on MacBook 2020 machines due to the following error:
```
** BUILD FAILED **

The following build commands failed:
	CompileC /Users/macpro13/o3de/osx_build/Code/Editor/O3DE.build/profile/EditorLib.build/Objects-normal/x86_64/LogFile_mac.o /Users/macpro13/o3de/Code/Editor/LogFile_mac.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'EditorLib' from project 'O3DE')
(1 failure)
```
The issue has reproduced on MacBook 2020 machines:
- MacBook Pro 2020 (Intel Core i5, Big Sur 11.3.1), XCode 13.2.1
- MacBook Air 2020 (M1, Monterey 12.0.1), XCode 13.2.1

The issue has not reproduced on the following machine:
- MacBook Pro 2017 (Intel Core i7, Big Sur 11.3.1), XCode 12.5

Please note that the issue occurs when building AutomatedTesting, as well as building the SDK layout.

**Steps to reproduce**
1. Create new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
2. Obtain python packages with get_python.sh script located in the o3de/python directory.
3. Create osx_build folder under the o3de location. 
4. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
```
5. Configure and generate the Xcode solution using this command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`
6. Build Mac binaries:
`cmake -E time cmake --build . --target ALL_BUILD --config profile -j 2`

**Expected behavior**
Building completes successfully.

**Actual behavior**
Building fails due to a command failure.

**Screenshot**
<img width="882" alt="BuildFailed_LogFile_mac" src="https://user-images.githubusercontent.com/86953659/157250401-ea831c36-f367-4663-b339-487f53ff07ec.png">

**Found in Branch**
Development (6df71ce)

**Desktop/Device**
- Device: MacBook Pro
- OS: macOS
- Version: Big Sur 11.3.1
- CPU: Intel Core i5 2.0GHz Quad-Core
- GPU: Intel Iris Plus Graphics 1536 MB
- Memory: 16GB

**Additional context**
[Build_FullLog.txt](https://github.com/o3de/o3de/files/8206680/Build_FullLog.txt)

**Comment by amzn-sj** [comment]
Fixed here: https://github.com/o3de/o3de/pull/8143

**Comment by LB-KacperKapusta** [comment]
Verified as fixed in Development (7a75e34).
